Controller/PositionableController.php

```diff
@@ -75,13 +76,18 @@ private function getEntity(CRUDElement $element, $id)
 
     /**
      * @param CRUDElement $element
+     * @param Request $request
      * @return RedirectResponse
      */
-    private function getRedirectResponse(CRUDElement $element)
+    private function getRedirectResponse(CRUDElement $element, Request $request)
     {
-        return new RedirectResponse(
-            $this->router->generate($element->getRoute(), $element->getRouteParameters())
-        );
+        if ($request->query->get('redirect_uri')) {
+            $uri = $request->query->get('redirect_uri');
```

Comment:
``redirect_uri`` parameter is by convension passed in query by admin-bundle's actions and batch actions, because it should never be part of any route's parameters. It's also pulled out from query by all admin-bundle's context handlers so why should we make an exception here?